src/documentation/programming-guide.md

```diff
@@ -464,8 +464,33 @@ tree, [2]
 
 Thus, `GroupByKey` represents a transform from a multimap (multiple keys to individual values) to a uni-map (unique keys to collections of values).
 
-> **A Note on Key/Value Pairs:** Beam represents key/value pairs slightly differently depending on the language and SDK you're using. In the Beam SDK for Java, you represent a key/value pair with an object of type `KV<K, V>`. In Python, you represent key/value pairs with 2-tuples.
+`CoGroupByKey` is a `PTransform` based on `GroupByKey`, differs from `GroupByKey`, it merges multiple key/value `PCollection`s with the same key type, and emits a collection of `KV<K, CoGbkResult>` pairs. It's usually used as a JOIN operation in Beam pipeline(describe in [design-your-pipeline#multiple-sources]({{ site.baseurl }}/documentation/pipelines/design-your-pipeline/#multiple-sources)). 
```

Comment:
Your additions would be great as a subsection in Using GroupByKey, so it has a little more visibility. That would help it stand out from the basic GroupByKey text.
*##### Joins with CoGroupByKey*